### DISCUSSION TITLE
Using vectors in a model

### DISCUSSION THREAD

**Discussion by pascalschulthess** [nlmixr2/rxode2#710] [Q&A] [answered]
I'm very new to RxODE and nlmixr so forgive me if this is something obvious and long known.
Is it not possible to reference elements of vectors in nlmixr2 models? At least I got the following error:
```Error: rxode2 syntax error:
:004: x[1] <- 5
       ^
more errors could be listed above
```
Thanks

**Comment by billdenney** [comment, marked as answer]
I think that is not possible.  But, it's also unclear what your goal is.  Can you share what you're trying to do and how the vector is helping?  Perhaps, can you share your model or an example model where you need that?

**Reply by pascalschulthess** [reply]
Thanks for the swift response. I'm implementing a minimal PBPK model that contains the two pore model and I thought that I could use vectors to calculate small and large pore parameters.
I'm quite surprised to hear that vectors are not possible in RxODE. But, I'll find another way.

**Reply by billdenney** [reply]
The typical method would be to use a parameter like `pore_large` and `pore_small` rather than `x[1]` and `x[2]`.

If you can give a more complete example, I could possibly give a better solution.

**Reply by pascalschulthess** [reply]
Thanks. I can expand this but I thought I could save some lines.
Any idea why vectors are not allowed? Or if statements for that matter?

**Reply by mattfidler** [reply]
`if` statements are allowed so are `while` statements.

If you are trying to save lines you may have thought to do something like:

`x[1:10] = 20`  

This is not implemented at this point because all the different way of handling R expressions would have to be built into the c-based parser and expanded to be able to be used in the derivative handling required for `nlmixr2` optimization.  It would also have to be built into `lotri` for initial conditions.

If you have a question of things that are supported in the `rxode2` syntax you can look at the following vignette:

https://nlmixr2.github.io/rxode2/articles/rxode2-syntax.html
